### PR TITLE
Use the collectible interface, which makes use of comparable and of t…

### PR DESCRIPTION
**Pull request by floppyzedolfin**
…he new Sortable interface

### CODE REVIEW

**Comment by floppyzedolfin** [comment]
I don't know if this is a good idea. I feel we're drifting away from the generics and into the interfaces.

File: chapter-07/final/collectors/load.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 )
 
 // Load reads the file and returns the list of collectors, and their beloved books, found therein.
-func Load[T comparable](filePath string) (Collectors[T], error) {
+func Load[T collectible](filePath string) (Collectors[T], error) {
```
Comment (ablqk) [inline chapter-07/final/collectors/load.go:10]:
```suggestion
func Load[T any](filePath string) (Collectors[T], error) {
```

File: chapter-07/final/collectors/collector.go [hunk 2]
```diff
@@ -32,6 +46,13 @@ func (c Collectors[T]) FindCommon() []T {
 	return sortItems(itemsInCommon)
 }
 
+// Display prints the collectible items nicely.
+func Display[T collectible](w io.Writer, items []T) {
```
Comment (ablqk) [inline chapter-07/final/collectors/collector.go:50]:
```suggestion
func Display[T any](w io.Writer, items []T) {
```

File: chapter-07/final/collectors/collector.go [hunk 3]
```diff
@@ -32,6 +46,13 @@ func (c Collectors[T]) FindCommon() []T {
...
+	for _, item := range items {
+		_, _ = fmt.Fprintf(w, "- %s\n", item)
```
Comment (ablqk) [inline chapter-07/final/collectors/collector.go:52]:
```suggestion
		_, _ = fmt.Fprintf(w, "- %v\n", item)
```

File: chapter-07/final/collectors/load.go (hunk 1 above)
Comment (floppyzedolfin) [reply chapter-07/final/collectors/load.go:10]: can't - we need to return a Collectors[T]

**Pull request merged**